common/environment/environment.cpp

```diff
@@ -1336,12 +1536,221 @@ void CLocalEnvironment::clearCache()
         p.setown(conn->getRoot());
     }
     cache.kill();
-    machineCacheBuilt = false;
-    dropZoneCacheBuilt = false;
+    numOfDropzonesByComputer.clear();
+    init();
     resetPasswordsFromSDS();
 }
 
+IConstDropZoneInfo * CLocalEnvironment::findDropZoneByAddress(const char * netaddress, const char *targetFilePath) const
+{
+    IConstDropZoneInfo * retVal = NULL;
+
+    Owned<IConstMachineInfo> machineInfo = getMachineByAddress(netaddress);
+    if (!machineInfo)
+        return retVal;
+
+    SCMStringBuffer machineName;
+    machineInfo->getName(machineName);
+    const char * pmachineName = machineName.str();
+    Owned<IConstDropZoneInfoIterator> zoneIt= getDropZoneIteratorByComputer(pmachineName);
+    LOG(MCdebugInfo, unknownJob, "Check remote drop zones (%d) by iterator on '%s':", zoneIt->count(), pmachineName);
```

Comment:
Logging once per call, may be ok, but we don't elsewhere.
Think should be removed (of in a DEBUG block only)